From c07771114b0f6bdaec9971671fe1af4a87f99b86 Mon Sep 17 00:00:00 2001
From: Rob Clark <robclark@freedesktop.org>
Date: Sat, 12 Oct 2013 12:21:49 -0400
Subject: [PATCH 7/8] modetest: alpha buffers

Make upper-left corner for RGB32 buffers translucent, for testing
blending of AR24 vs XR24.

Signed-off-by: Rob Clark <robclark@freedesktop.org>
Signed-off-by: Sean Paul <seanpaul@chromium.org>
---
 tests/modetest/buffers.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/modetest/buffers.c b/tests/modetest/buffers.c
index 8206ce3..28bbd31 100644
--- a/tests/modetest/buffers.c
+++ b/tests/modetest/buffers.c
@@ -852,10 +852,11 @@ fill_tiles_rgb32(const struct format_info *info, unsigned char *mem,
 			div_t d = div(x+y, width);
 			uint32_t rgb32 = 0x00130502 * (d.quot >> 6)
 				       + 0x000a1120 * (d.rem >> 6);
+			uint32_t alpha = ((y < height/2) && (x < width/2)) ? 127 : 255;
 			uint32_t color =
 				MAKE_RGBA(rgb, (rgb32 >> 16) & 0xff,
 					  (rgb32 >> 8) & 0xff, rgb32 & 0xff,
-					  255);
+					  alpha);
 
 			((uint32_t *)mem)[x] = color;
 		}
-- 
1.9.0.rc1.175.g0b1dcb5

